File: Data Structures & Algorithms/concatenation-of-array/submission-4.cpp
class Solution {
public:
    vector<int> getConcatenation(vector<int>& nums) {
        vector<int> array(nums.size() * 2);
        int pointer = 0;
        for (int i = 0; i < array.size(); i++) {
            array[i] = nums[pointer];

            if (pointer < nums.size() - 1) {
                pointer ++;
            }
            else {
                pointer = 0;
            }


        }
        return array;
    }
};

File: Data Structures & Algorithms/top-k-elements-in-list/submission-4.py
class Solution:
    def topKFrequent(self, nums: List[int], k: int) -> List[int]:
        bucket = [[] for _ in range(len(nums) + 1)]
        frequency_count = {}

        for num in nums:
            frequency_count[num] = frequency_count.get(num, 0) + 1

        for key, value in frequency_count.items():
            bucket[value].append(key)

        res = []
        for i in range(len(bucket) - 1, 0, -1):
            if bucket[i]:
                for value in bucket[i]:
                    res.append(value)
                    if len(res) == k:
                        return res
